examples/gno.land/p/moul/mdlist/mdlist.gno

```diff
@@ -0,0 +1,83 @@
+// Package mdlist is a proof-of-concept for composable UIs patterns.
+//
+// It allows specifying a data source and then dynamically rendering it as
+// a list or a grid.
+package mdlist
+
+import (
+	"strings"
+
+	"gno.land/p/moul/md"
+)
+
+// XXX: configurable length UL
+// XXX: moul/pager.Pager-powered auto mdlist
+// XXX: func (l List) Render(path string) string {
+// XXX: add .Table()
+
+var defaultEncoder = NewEncoder()
+
+type List []Entry
+
+type Entry struct {
+	// Title is the title of the entry. Required.
+	Title string
+	// Body is the body of the entry. Required.
+	Body string
+	// Link is the URL of the entry. Optional.
+	Link string
+	// Links is a list of additional URLs of the entry. Optional.
+	Links []string
+}
+
+func (l List) UL() string {
+	items := make([]string, len(l))
+	for i, entry := range l {
+		title := truncateText(entry.Title, 40, false)
+		if entry.Link != "" {
+			title = md.Link(title, entry.Link)
+		}
+		body := truncateText(entry.Body, 80, true)
+		links := strings.Join(entry.Links, " ")
+		item := md.Bold(title) + ": " + md.Italic(body) + " " + links
+		items[i] = strings.TrimSpace(item)
+	}
+	return md.BulletList(items)
+}
+
+func (l List) Grid() string {
```

Comment:
Still on naming: `Grid` seems ambiguous to me, and I'd rename it to `Columns` for more clarity.

Reply:
8f189d3a4